contracts/modules/Payments.sol

```diff
@@ -60,17 +60,16 @@ abstract contract Payments is RouterImmutables {
         if (bips == 0 || bips > 10_000) revert InvalidBips();
         if (token == Constants.ETH) {
             uint256 balance = address(this).balance;
-            uint256 amount = (balance * bips) / FEE_BIPS_BASE;
-            uint256 fixedAmount = (amount * PAYMENT_AMOUNT_BIPS) / FEE_BIPS_BASE;
-            PAYMENT_RECIPIENT.safeTransferETH(fixedAmount);
-            recipient.safeTransferETH(amount - fixedAmount);
+            // store intermediate value to protect against phantom overflow
+            uint256 numerator = balance * bips;
```

Comment:
How does storing intermediate value protect against phantom overflow?
In both below cases, revert occurs when `balance * bips` > max uint256.
```
uint256 amount = (balance * bips) / FEE_BIPS_BASE;


uint256 numerator = balance * bips;
uint256 amount = numerator / FEE_BIPS_BASE;
```